entities/params/requestbody.go

```diff
@@ -1,8 +1,31 @@
 package params
 
-import "github.com/go-playground/validator/v10"
+import (
+	"github.com/go-playground/validator/v10"
+	"github.com/sirupsen/logrus"
 
+	myvalidator "github.com/mailbadger/app/validator"
+)
+
+// RequestBody represents type for request body structures to simplify validation
 type RequestBody interface {
-	trimSpaces()
-	Validate(v validator.Validate) (string, bool)
-}
\ No newline at end of file
+	TrimSpaces()
+}
+
+// Validate is generic function for validation request body
+func Validate(body RequestBody, v *validator.Validate) *myvalidator.FieldErrors {
```

Comment:
@Gligor23 I'm thinking we should return `error` here and have a custom ValidationError structure that implements the `Error` interface. That way it will be standardized way of returning errors and you can cast it, like so:

Let's say we have a struct like this:

```go
type ValidationError struct {
  Message string `json:"message"`
  Errors map[string]string `json:"errors"`
  ValidationErrors validator.ValidationErrors
}

func (e *ValidationError) Error() string {
    return e.ve.Error()
```

Now we can use it in Validate like so:
```go
func Validate(body RequestBody, v *validator.Validate) error {
  	body.TrimSpaces()

	// Validate the instance
	if err := v.Struct(body); err != nil {
		if fieldErrors, ok := err.(validator.ValidationErrors); ok {
			return &myvalidator.ValidationError{
                         Message: "Invalid parameters.",
                         Errors: map[string]string{}, // iterate the validation errors and make a map
                          ValidationErrors: err,
                        }
		}
		return &myvalidator.ValidationErrors{
                  Message: "Invalid parameters.",
                }
	}

	return nil
}
```

We could also move this `Validate` function to the `validator` package and we won't need the 2nd argument then. It'll just be `validator.Validate(body)`